src/ports/postgres/modules/utilities/utilities.py_in

```diff
@@ -709,16 +709,35 @@ def _check_groups(tbl1, tbl2, grp_list):
     return ' AND '.join([" {tbl1}.{i} = {tbl2}.{i} ".format(**locals())
                          for i in grp_list])
 
-
-def _grp_from_table(tbl, grp_list):
-    """
-    Helper function for selecting grouping columns of a table
+def get_filtered_cols_subquery_str(include_from_table, exclude_from_table,
+                                   filter_cols_list):
+    """
+    This function returns a subquery string with columns in the filter_cols_list
```

Comment:
I'm confused with this description string. My understanding is that the function doesn't really filter the columns - it filters the values for the columns i.e. it returns a subquery string that would filter values in exclude_from_table. If you're interested in filtering columns, then wouldn't querying the catalog and then doing a set difference be the better option?